### PR TITLE
Added color ramp selection and opacity slider for layers

### PR DESCRIPTION
**Pull request by ZachSand**
I've added a color ramp selector drop down and an opacity slider that respectively updates the current layer. The color ramp appends a string onto the tms url that is parsed before rendering and the opacity slider takes advantage of leaflet's setOpacity function for a layer. 

The Slider.jsx component, how I see it, is a React wrapper around the [bootstrap-slider](https://github.com/seiyria/bootstrap-slider). There were issues trying to get everything to function correctly with just the native bootstrap slider. Notice, an additional dependency has been added -- the bootstrap slider. 

The color ramp images were pulled from the admin project and the ColorRampMap was pulled from geotrellis/services.

Known (possible) defect of color ramp selector, steps to reproduce: 
- Run the server, select a ramp from the dropdown besides the first one, the ramp scale image changes. Press button to select an "ingested" layer, the color ramp that is shown for the selected layer is not the same as the color ramp chosen from the drop down and displayed in the scale. This only happens if a color other than the first is selected before any layer has **ever** been rendered.

### CODE REVIEW

File: viewer/src/scripts/components/LeafletMap.jsx [hunk 1]
```diff
@@ -40,47 +50,102 @@ var LeafletMap = React.createClass({
...
-      var args = active.band;      
-      args['breaks'] = active.entry.breaks.join(',');      
+     
+      var args = active.band; 
+      args['breaks'] = active.entry.breaks.join(',');
+
+      if(this.layer) {
+        var tmsUrl = this.layer._url
+        var findColorRegex = /&colorRamp=/;
+        tmsUrlSplit = tmsUrl.split(findColorRegex);
```
Comment (mmccartney) [inline viewer/src/scripts/components/LeafletMap.jsx]: this is a bit of a kludge and probably should be commented as such.  it's best to use a library to parse the URL and extract the parameter.  for example, this code breaks if the first query param was `colorRamp`:  `http://.../foo?colorRamp=`

File: viewer/src/scripts/components/LeafletMap.jsx [hunk 2]
```diff
@@ -40,47 +50,102 @@ var LeafletMap = React.createClass({
...
       this.layer = newLayer;
-
       _.delay(removeLayer, 1000, this.map, oldLayer);
+  },
 
+  handleUpdateColorRamp: function() {
+    var active = this.props.active; 
+    var tmsUrl = this.layer._url
+    var findColorRegex = /&colorRamp/;
+    var tmsUrlSplit = tmsUrl.split(findColorRegex);
```
Comment (mmccartney) [inline viewer/src/scripts/components/LeafletMap.jsx]: another kludge here which potentially breaks other query parameters if they come after `colorRamp` on the URL.  this only works as desired if `colorRamp` is the last query parameter on the URL.

**Comment by mmccartney** [comment]
@lossyrob have you had a chance to look at this?  Looks reasonable to me although I flagged a couple of kludges.  Otherwise looks like a decent contribution

**Comment by lossyrob** [comment]
For sure. I have to do some merging work, which I'm going to do later on tonight. but these 2 PR's are going to be merged in.

File: server/src/main/scala/geotrellis/admin/server/CatalogService.scala [hunk 3]
```diff
@@ -94,15 +94,21 @@ object CatalogService extends ArgApp[CatalogArgs] with SimpleRoutingApp with Cor
                    sys.error("NOPE!")
                  }
                }
-
+            
+             var rampOption = "blue-to-yellow-to-red-heatmap";  //first ramp in drop down
```
Comment (lossyrob) [inline server/src/main/scala/geotrellis/admin/server/CatalogService.scala]: always use `val` in scala unless you really want something mutable

Comment (lossyrob) [inline server/src/main/scala/geotrellis/admin/server/CatalogService.scala]: e.g.:
```scala
              val rampOption =
                colorOption match {
                  case Some(colorRamp) =>
                    colorRamp
                  case None =>
                    "blue-to-yellow-to-red-heatmap" //first ramp in drop down
                }
```

Comment (lossyrob) [inline server/src/main/scala/geotrellis/admin/server/CatalogService.scala]: or,

```scala
              // Use the user specified color ramp, or default to blue to yellow to red.
              val rampOption = 
                colorOption.getOrElse("blue-to-yellow-to-red-heatmap")
```

File: server/src/main/scala/geotrellis/admin/server/CatalogService.scala [hunk 4]
```diff
@@ -94,15 +94,21 @@ object CatalogService extends ArgApp[CatalogArgs] with SimpleRoutingApp with Cor
...
+              case Some(colorRamp) => 
+                rampOption = colorRamp
+              case None =>
+             }  
 
              breaksOption match {
                case Some(breaks) =>
-                 if(layer == "diff") tile.renderPng(ColorRamps.LightToDarkGreen, breaks.split(",").map(_.toInt)).bytes
-                 else tile.renderPng(ColorRamps.BlueToOrange, breaks.split(",").map(_.toInt)).bytes
+                 var colorToRender = ColorRampMap.getOrElse(rampOption, ColorRamps.HeatmapBlueToYellowToRedSpectrum)
```
Comment (lossyrob) [inline server/src/main/scala/geotrellis/admin/server/CatalogService.scala]: If you defaulted above you shouldn't default here.

**Comment by lossyrob** [comment]
I get an error: RROR in ./src/scripts/components/LeafletMap.jsx
Module not found: Error: Cannot resolve module 'bootstrap-slider/dist/css/bootstrap-slider.css' in /Users/rob/proj/gt/gt-admin-codereview/viewer/src/scripts/components
 @ ./src/scripts/components/LeafletMap.jsx 16:0-63

ERROR in ./src/scripts/components/LeafletMap.jsx
Module not found: Error: Cannot resolve module 'bootstrap-slider' in /Users/rob/proj/gt/gt-admin-codereview/viewer/src/scripts/components
 @ ./src/scripts/components/LeafletMap.jsx 13:22-49

ERROR in ./src/scripts/components/Slider.jsx
Module not found: Error: Cannot resolve module 'bootstrap-slider' in /Users/rob/proj/gt/gt-admin-codereview/viewer/src/scripts/components
 @ ./src/scripts/components/Slider.jsx 11:22-49

I'm going to make a pull request to your branch which you can merge, that will update your branch to master. Please add any missing files and let me know when it's ready.

**Comment by ZachSand** [comment]
Did you run npm install? I updated the package.json to include the dependency.

**Comment by ZachSand** [comment]
The following changes seem to work, however I'm unfamiliar with Scala's Option class. Is there a better way to get the value from the Option than using ".head"?

    val rampOption = 
      colorOption.getOrElse("blue-to-yellow-to-red-heatmap") 

    breaksOption match {
      case Some(breaks) =>
        tile.renderPng(ColorRampMap.get(rampOption).head, breaks.split(",").map(_.toInt)).bytes

**Comment by lossyrob** [comment]
You were right about `npm install`

PR looks good. I made some updates so am closing that.

To answer your question about `Option.head`, call `head` will only work if there is actually something there, and if the `rampOption` is `None`, an exception would be thrown.

**Pull request closed without merging**